pkg/webui/console/views/device-general-settings/index.js

```diff
@@ -187,9 +200,12 @@ export default class DeviceGeneralSettings extends React.Component {
     // 2. Disable the section if the device is OTAA and joined since no fields are stored in the AS.
     // 3. Disable the section if NS is not in cluster, since activation mode is unknown.
     // 4. Disable the seciton if `application_server_address` is not equal to the cluster AS address.
+    // 5. Disable the section if the user has no appropriate rights to edit keys.
     const sameAsAddress = getComponentBaseUrl(asConfig) === device.application_server_address
     const isJoined = isDeviceJoined(device)
-    const asDisabled = !asEnabled || (isOTAA && !isJoined) || !nsEnabled || !sameAsAddress
+    const showKeys = mayReadKeys && mayEditKeys
```

Comment:
- If a user is allowed to read keys, but not write them, we can show the keys as read-only
- If a user is allowed to write keys, but not read them, we can still let them reset the keys

Reply:
> If a user is allowed to read keys, but not write them, we can show the keys as read-only

The users can see the keys in the device overview page if they have the read right. the only reason we show the `session.dev_addr` field and disable it in the AS section is for cases when NS not in cluster. however, now it seems redundant

> If a user is allowed to write keys, but not read them, we can still let them reset the keys

Shall we inform the user that the key might be there it just not visible because of insufficient rights? Asking because imo an empty field can make the user think that the key (`session.keys.app_s_key.key`) is not set, while him/her just not having the rights to view the key. Note, that `app_s_key` is not required by AS

Reply:
> ...we could render an empty input together with a warning that values cannot be viewed and new values potentially overwrite existing ones.

I dont think we should not do this. Such a combination of rights (no read, but write) is applicable mostly to API keys, not users. Hence I dont think we should invest time and make the implementation more complex to support such edge case. I would like to hear from @htdvisser and @johanstokking regarding the notification.

The main purpose of this PR is to avoid errors when submitting without the rights as well as still fetching js/ns fields without the keys if the user is not allowed to fetch them.

Other implementation specific questions if we decide to add the notification and handle the case when the user is able to edit but not read the keys:
1. Do we still generate the keys automatically if the user submits the form with empty key fields?
2. What is the message we want to show in the notification? Asking because any wording I can come up with is very stack/implementation specific and I guess we dont want to bother the users with this.

Reply:
> Hence I dont think we should invest time and make the implementation more complex to support such edge case.

I agree.

> I would like to hear from @htdvisser and @johanstokking regarding the notification.

>What is the message we want to show in the notification? 

Which notification exactly?

> 1. Do we still generate the keys automatically if the user submits the form with empty key fields?

No, this should be an explicit action/checkbox

Reply:
> Such a combination of rights (no read, but write) is applicable mostly to API keys, not users

I wouldn't assume that. Maybe I want to allow an entire org to create devices with keys, but don't read them after create.

> Do we still generate the keys automatically if the user submits the form with empty key fields?

I've always been against this. I think we just need a 🔁 button to generate keys.

> What is the message we want to show in the notification?

Perhaps something like "You can not read this value, but you can overwrite it if you want"